src/lib.rs

```diff
@@ -885,51 +925,747 @@ impl GooseAttack {
+                });
+            }
+
+            if self.configuration.no_task_metrics {
+                return Err(GooseError::InvalidOption {
+                    option: "--no-task-metrics".to_string(),
+                    value: self.configuration.no_task_metrics.to_string(),
+                    detail:
+                        "The --no-task-metrics flag can not be set together with the --worker flag."
+                            .to_string(),
+                });
+            }
+
+            if self.configuration.only_summary {
+                return Err(GooseError::InvalidOption {
+                    option: "--only-summary".to_string(),
+                    value: self.configuration.only_summary.to_string(),
+                    detail:
+                        "The --only-summary flag can not be set together with the --worker flag."
+                            .to_string(),
+                });
+            }
+
+            if self.configuration.status_codes {
+                return Err(GooseError::InvalidOption {
+                    option: "--status-codes".to_string(),
+                    value: self.configuration.status_codes.to_string(),
+                    detail:
+                        "The --status-codes flag can not be set together with the --worker flag."
+                            .to_string(),
+                });
+            }
+
+            if self.configuration.no_reset_metrics {
+                return Err(GooseError::InvalidOption {
+                    option: "--no-reset-metrics".to_string(),
+                    value: self.configuration.no_reset_metrics.to_string(),
+                    detail: "The --no-reset-metrics flag can not be set together with the --worker flag.".to_string(),
+                });
+            }
+
+            if self.configuration.no_hash_check {
+                return Err(GooseError::InvalidOption {
+                    option: "--no-hash-check".to_string(),
+                    value: self.configuration.no_hash_check.to_string(),
+                    detail:
+                        "The --no-hash-check flag can not be set together with the --worker flag."
+                            .to_string(),
+                });
+            }
+        }
+
+        // Otherwise run in standalone attack mode.
+        if self.attack_mode == GooseMode::Undefined {
+            self.attack_mode = GooseMode::StandAlone;
+
+            if self.configuration.no_hash_check {
+                return Err(GooseError::InvalidOption {
+                    option: "--no-hash-check".to_string(),
+                    value: self.configuration.no_hash_check.to_string(),
+                    detail: "The --no-hash-check flag can not be set without also setting the --manager flag.".to_string(),
+                });
+            }
+        }
+
+        Ok(())
+    }
+
+    // Determine how many workers to expect.
+    fn set_expect_workers(&mut self) -> Result<(), GooseError> {
```

Comment:
@LionsAd In this case we overload the configuration array (and I suspect this is the preferred way to handle defaults).